jetty-websocket/javax-websocket-server-impl/src/main/java/org/eclipse/jetty/websocket/jsr356/server/ServerContainer.java

```diff
@@ -191,35 +195,39 @@ protected void doStart() throws Exception
 
     public ServerEndpointMetadata getServerEndpointMetadata(final Class<?> endpoint, final ServerEndpointConfig config) throws DeploymentException
     {
-        ServerEndpointMetadata metadata = null;
-
-        ServerEndpoint anno = endpoint.getAnnotation(ServerEndpoint.class);
-        if (anno != null)
-        {
-            // Annotated takes precedence here
-            AnnotatedServerEndpointMetadata ametadata = new AnnotatedServerEndpointMetadata(this, endpoint, config);
-            AnnotatedEndpointScanner<ServerEndpoint, ServerEndpointConfig> scanner = new AnnotatedEndpointScanner<>(ametadata);
-            metadata = ametadata;
-            scanner.scan();
-        }
-        else if (Endpoint.class.isAssignableFrom(endpoint))
+        try
         {
-            // extends Endpoint
-            @SuppressWarnings("unchecked")
-            Class<? extends Endpoint> eendpoint = (Class<? extends Endpoint>)endpoint;
-            metadata = new SimpleServerEndpointMetadata(eendpoint, config);
+            ServerEndpointMetadata metadata;
+            ServerEndpoint anno = endpoint.getAnnotation(ServerEndpoint.class);
+            if (anno != null)
+            {
+                // Annotated takes precedence here
+                AnnotatedServerEndpointMetadata ametadata = new AnnotatedServerEndpointMetadata(this, endpoint, config);
+                AnnotatedEndpointScanner<ServerEndpoint, ServerEndpointConfig> scanner = new AnnotatedEndpointScanner<>(ametadata);
+                metadata = ametadata;
+                scanner.scan();
+            }
+            else if (Endpoint.class.isAssignableFrom(endpoint))
+            {
+                // extends Endpoint
+                @SuppressWarnings("unchecked")
+                Class<? extends Endpoint> eendpoint = (Class<? extends Endpoint>)endpoint;
+                metadata = new SimpleServerEndpointMetadata(eendpoint, config);
+            }
+            else
+            {
+                String err = "Not a recognized websocket [" + endpoint.getName() +
+                    "] does not extend @" + ServerEndpoint.class.getName() +
+                    " or extend from " + Endpoint.class.getName();
+                throw new DeploymentException(err);
+            }
+
+            return metadata;
         }
-        else
+        catch (RuntimeException e)
```

Comment:
Should not you catch `Throwable` here, to cover cases such as `NoClassDefFoundError`, etc.?

Reply:
I just didn't want to catch the `DeploymentException`s which may be thrown from within this block. But I can just re-throw them here if they are instance of `DeploymentException`.